src/gui/qgsattributeform.cpp

```diff
@@ -2204,28 +2204,16 @@ void QgsAttributeForm::updateJoinedFields( const QgsEditorWidgetWrapper &eww )
 
 bool QgsAttributeForm::fieldIsEditable( int fieldIndex ) const
 {
-  bool editable = false;
-
   if ( mLayer->fields().fieldOrigin( fieldIndex ) == QgsFields::OriginJoin )
   {
     int srcFieldIndex;
     const QgsVectorLayerJoinInfo *info = mLayer->joinBuffer()->joinForFieldIndex( fieldIndex, mLayer->fields(), srcFieldIndex );
 
     if ( info && !info->hasUpsertOnEdit() && mMode == QgsAttributeEditorContext::AddFeatureMode )
```

Comment:
Is this check needed in the new method too?

Reply:
Yes, in case info is null or not editable [here](https://github.com/troopa81/QGIS/blob/fix_multiedit_button_with_auxiliary_storage/src/core/qgsvectorlayerutils.cpp#L922).

Do you prefer several return? I just keep the method the way it was written in the first place.

Reply:
No, I mean I think it should be in QgsVectorLayerUtils::fieldIsEditable. If upsert isn't possible then the field can only be edited if the joined feature already exists, and this will impact the multi edit mode too.

Reply:
You mean like this f3f7a77 ?

So now, the multieditbutton is visible if all selected feature are not readOnly. That means that the visibility button change upon selected feature. Is that OK?

Once we agree on the behavior, I will add some unit test.